pyproject.toml

```diff
@@ -1,3 +1,78 @@
+[build-system]
+requires = ["poetry-core>=1.0.0", "poetry-dynamic-versioning"]
+build-backend = "poetry_dynamic_versioning.backend"
+
+[tool.poetry-dynamic-versioning]
+enable = true
+vcs = "git"
+style = "semver"
+pattern = "default-unprefixed"
+
+[tool.poetry.dependencies]
+python = "^3.7"
+
+# Python lack of functionalities from future versions
+importlib-metadata = { version = "*", python = "<3.8" }
+tensorflow = { version = ">=2.0.0", optional = true }
```

Comment:
What about `tensorflow-macos`? I remember that the "standard" `tensorflow` is not necessarily required to use TF in general as there is an alternative build (probably prepared mostly for MacOS on ARM - Apple Silicon). I'll try to check that on my Mac later so let's leave this comment to be aware of this issue potentially.

Reply:
I know it's set to optional but I'm not sure how does it impact this setup (no tf, mac m1, tf-macos package)

Reply:
I was thinking of releasing once pytorch, tensorboardx and tensorflow are in place and in that case we can use the code snippet you shared
```python
if find_spec("tensorflow") or find_spec("tensorflow-macos"):
    ...
if find_spec("tensorboardx"):
    ...
else:
    logger.warning("None of the supported packages was installed.")
```

Also, if there are multiple package names for tensorflow, does it make sense to just do `try..except` instead, so that even if new package name comes for tensorflow, we would be able to integrate.

Thanks!

Reply:
Sure. `try ... except` for tensorflow is probably a much more bulletproof solution.

Reply:
Done.